src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -25,17 +33,25 @@ public void calcAverageAgeOfEmployees() {
     public void findPersonWithLongestFullName() {
         List<Employee> employees = Example1.getEmployees();
 
-        Person expected = null;
+        Person expected = employees
+                .stream()
+                .max(Comparator.comparingInt(e -> e.getPerson().getFullName().length())).get().getPerson();
```

Comment:
you'd better first map stream of employees to stream of persons.
Also use orElseThrow or orElse methods instead of get.

Reply:
Why is it better? It's just additional traverse.
And it is just study exercise and that's why I used get.

Reply:
You can count method calls.  Your code takes as input a stream of Employees  ( let it has length N)  and for each employee converts it to person while finding  max fullname. And then converts to person one more time  to get the result.  so N+1 method calls.

But we can just map stream to the persons stream in order to work with more simple and lightweihgt objects. And it forces us to make only N  getPerson calls.

Another cause - you just simplify  extraction of int value for comparator and make your code more readable.

As I know jvm simplifies chain operations by itself, and you can write 
  filter( simple condition). filter( simple condition)  instead of  filter ( complex condition)

According to get method. I think it's just more friendly.

Reply:
Yes, you're right! It works faster.